app/browser/api/ledger.js

```diff
@@ -104,13 +102,264 @@ const fileTypes = {
+// Notifications
+const sufficientBalanceToReconcile = (state) => {
+  const balance = Number(ledgerState.getInfoProp(state, 'balance') || 0)
+  const unconfirmed = Number(ledgerState.getInfoProp(state, 'unconfirmed') || 0)
+  const btc = ledgerState.getInfoProp(state, 'btc')
+  return btc && (balance + unconfirmed > 0.9 * Number(btc))
+}
+const hasFunds = (state) => {
+  const balance = getSetting(settings.PAYMENTS_ENABLED)
+    ? Number(ledgerState.getInfoProp(state, 'balance') || 0)
+    : 0
+  return balance > 0
+}
+const shouldShowNotificationReviewPublishers = () => {
+  const nextTime = getSetting(settings.PAYMENTS_NOTIFICATION_RECONCILE_SOON_TIMESTAMP)
+  return !nextTime || (new Date().getTime() > nextTime)
+}
+const shouldShowNotificationAddFunds = () => {
+  const nextTime = getSetting(settings.PAYMENTS_NOTIFICATION_ADD_FUNDS_TIMESTAMP)
+  return !nextTime || (new Date().getTime() > nextTime)
+}
+const notifications = {
+  text: {
+    hello: locale.translation('updateHello'),
+    paymentDone: undefined,
+    addFunds: locale.translation('addFundsNotification'),
+    tryPayments: locale.translation('notificationTryPayments'),
+    reconciliation: locale.translation('reconciliationNotification'),
+    walletConvertedToBat: locale.translation('walletConvertedToBat')
+  },
+  pollingInterval: 15 * miliseconds.minutes,
+  timeout: undefined,
+  displayOptions: {
+    style: 'greetingStyle',
+    persist: false
+  },
+  init: (state) => {
+    // Check if relevant browser notifications should be shown every 15 minutes
+    if (notifications.timeout) {
+      clearInterval(notifications.timeout)
+    }
+    notifications.timeout = setInterval((state) => {
+      notifications.onInterval(state)
+    }, notifications.pollingInterval, state)
+
+    // Show relevant browser notifications on launch
+    notifications.onLaunch(state)
+  },
+  onLaunch: (state) => {
+    if (!getSetting(settings.PAYMENTS_NOTIFICATIONS)) {
+      return
+    }
+    // Show one-time BAT conversion message:
+    // - if payments are enabled
+    // - user has a positive balance
+    // - this is an existing profile (new profiles will have firstRunTimestamp matching btcToBatTimestamp)
+    // - notification has not already been shown yet
+    // (see https://github.com/brave/browser-laptop/issues/11021)
+    if (hasFunds(state)) {
+      const isNewInstall = state.get('firstRunTimestamp') === state.getIn(['migrations', 'btcToBatTimestamp'])
+      const hasBeenNotified = state.getIn(['migrations', 'btcToBatTimestamp']) !== state.getIn(['migrations', 'btcToBatNotifiedTimestamp'])
+      if (!isNewInstall && !hasBeenNotified) {
+        notifications.showBraveWalletUpdated()
+      }
+    }
+  },
+  onInterval: (state) => {
+    if (!getSetting(settings.PAYMENTS_NOTIFICATIONS)) {
+      return
+    }
+    if (getSetting(settings.PAYMENTS_ENABLED)) {
+      notifications.showEnabledNotifications(state)
+    } else {
+      notifications.showDisabledNotifications(state)
```

Comment:
before we showed this notification only when `!getSetting(settings.PAYMENTS_NOTIFICATIONS)` is that change intentional?

Reply:
logic shouldn't have changed at all (just moved). If PAYMENT_NOTIFICATIONS is set, none of the notifications should be shown

Reply:
I guess I'm thinking about getSetting(settings.PAYMENTS_NOTIFICATIONS) is false but getSetting(settings.PAYMENTS_ENABLED is true, before it would show.

Reply:
Great catch- I rolled back to the original behavior (so that there is no change from previous code)